src/main/java/appeng/core/sync/packets/PacketNEIRecipe.java

```diff
@@ -62,117 +61,297 @@
+	// {
+	// IContainerCraftingPacket cct = (IContainerCraftingPacket) con;
+	// IGridNode node = cct.getNetworkNode();
+	// if( node != null )
+	// {
+	// IGrid grid = node.getGrid();
+	// if( grid == null )
+	// {
+	// return;
+	// }
+	//
+	// IStorageGrid inv = grid.getCache( IStorageGrid.class );
+	// IEnergyGrid energy = grid.getCache( IEnergyGrid.class );
+	// ISecurityGrid security = grid.getCache( ISecurityGrid.class );
+	// IInventory craftMatrix = cct.getInventoryByName( "crafting" );
+	// IInventory playerInventory = cct.getInventoryByName( "player" );
+	//
+	// Actionable realForFake = cct.useRealItems() ? Actionable.MODULATE : Actionable.SIMULATE;
+	//
+	// if( inv != null && this.recipeStack != null && security != null )
+	// {
+	// InventoryCrafting testInv = new InventoryCrafting( new ContainerNull(), 3, 3 );
+	// for( int x = 0; x < 9; x++ )
+	// {
+	// if( this.recipeStack[x] != null && this.recipeStack[x].length > 0 )
+	// {
+	// testInv.setInventorySlotContents( x, this.recipeStack[x][0] );
+	// }
+	// }
+	//
+	// IRecipe r = Platform.findMatchingRecipe( testInv, pmp.worldObj );
+	//
+	// if( r != null && security.hasPermission( player, SecurityPermissions.EXTRACT ) )
+	// {
+	// ItemStack is = r.getCraftingResult( testInv );
+	//
+	// if( is != null )
+	// {
+	// IMEMonitor<IAEItemStack> storage = inv.getItemInventory();
+	// IItemList all = storage.getStorageList();
+	// IPartitionList<IAEItemStack> filter = ItemViewCell.createFilter( cct.getViewCells() );
+	//
+	// for( int x = 0; x < craftMatrix.getSizeInventory(); x++ )
+	// {
+	// ItemStack patternItem = testInv.getStackInSlot( x );
+	//
+	// ItemStack currentItem = craftMatrix.getStackInSlot( x );
+	// if( currentItem != null )
+	// {
+	// testInv.setInventorySlotContents( x, currentItem );
+	// ItemStack newItemStack = r.matches( testInv, pmp.worldObj ) ? r.getCraftingResult( testInv ) : null;
+	// testInv.setInventorySlotContents( x, patternItem );
+	//
+	// if( newItemStack == null || !Platform.isSameItemPrecise( newItemStack, is ) )
+	// {
+	// IAEItemStack in = AEItemStack.create( currentItem );
+	// if( in != null )
+	// {
+	// IAEItemStack out = realForFake == Actionable.SIMULATE ? null : Platform.poweredInsert( energy, storage, in,
+	// cct.getSource() );
+	// if( out != null )
+	// {
+	// craftMatrix.setInventorySlotContents( x, out.getItemStack() );
+	// }
+	// else
+	// {
+	// craftMatrix.setInventorySlotContents( x, null );
+	// }
+	//
+	// currentItem = craftMatrix.getStackInSlot( x );
+	// }
+	// }
+	// }
+	//
+	// // True if we need to fetch an item for the recipe
+	// if( patternItem != null && currentItem == null )
+	// {
+	// // Grab from network by recipe
+	// ItemStack whichItem = Platform.extractItemsByRecipe( energy, cct.getSource(), storage, player.worldObj, r, is,
+	// testInv, patternItem, x, all, realForFake, filter );
+	//
+	// // If that doesn't get it, grab exact items from network (?)
+	// // TODO see if this code is necessary
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)